web/src/pages/Main.test.tsx

```diff
@@ -0,0 +1,118 @@
+import { fireEvent, render, screen } from '@testing-library/react'
+import '@testing-library/jest-dom/extend-expect'
+import { MainPage } from './Main'
+import MercuryAPI from '../api/MercuryAPI'
+import { ComponentResponse, MultiflashResponse } from '../api/generated'
+import { AxiosResponse } from 'axios'
+
+test('renders without crashing and calculate run populates tables', async () => {
+  const mockMercuryApi = new MercuryAPI()
+  const multiflashResponse: MultiflashResponse = {
+    phaseValues: {
+      Vapor: {
+        percentage: 0.9530398821754837,
+        mercury: 9576.977265669584,
+      },
+      Aqueous: {
+        percentage: 0.04596219288894959,
+        mercury: 42.95333958603203,
+      },
+      Mercury: {
+        percentage: 0.0009979249355665074,
+        mercury: 1000000000,
+      },
+    },
+    componentFractions: {
+      '1': [0.0487818478493959, 0.00003178625137356113, 9.357933137511921e-24],
+      '2': [0.004926658586182084, 5.606370638941071e-8, 9.521979777029961e-25],
+      '3': [0.016978518176734285, 0.999923029961448, 3.2368004598608978e-24],
+      '5': [0.0000011288993580273517, 3.858025869444538e-9, 1.0000000000000002],
+      '101': [
+        0.6851613143677305, 0.00002077078423705684, 1.319319561831917e-22,
+      ],
+      '201': [
+        0.09046902819956745, 0.000004073890325362129, 1.7297094356248018e-23,
+      ],
+      '301': [
+        0.047759949263746916, 0.000007530068352386529, 9.075924729973352e-24,
+      ],
+      '401': [
+        0.007944465811905398, 7.791208565250264e-7, 1.5028484039426094e-24,
+      ],
+      '402': [
+        0.016227467901725096, 0.0000024012881239653258, 3.0664046159226887e-24,
+      ],
+      '503': [
+        0.005438171625943571, 4.946930604181399e-7, 1.0230429071522505e-24,
+      ],
+      '504': [
+        0.006415100362966198, 8.620089341914816e-7, 1.2054916829998997e-24,
+      ],
+      '605': [
+        0.06989634895474478, 0.000008212011556262026, 1.3063098591813244e-23,
+      ],
+    },
+    feedMolecularWeight: 26.494307395,
+  }
+
+  const components: ComponentResponse = {
```

Comment:
This is the second time we initiate ComponentResponse in test. Could we instead make it a shareable variable?

Reply:
Possible. Out of scope for this PR. Creating a separate issue for this.